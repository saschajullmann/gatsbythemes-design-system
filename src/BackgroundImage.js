import styled from '@emotion/styled'
import PropTypes from 'prop-types'
import { width } from 'styled-system'
import theme from './theme'

const image = props =>
  props.image ? { backgroundImage: `url(${props.image})` } : null

const height = props => (props.height ? { height: props.height } : null)

const BackgroundImage = styled.div`
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  background-color: ${props => props.theme.colors.gray};
  ${image} ${height} ${width};
`
const numberStringOrArray = PropTypes.oneOfType([
  PropTypes.number,
  PropTypes.string,
  PropTypes.array
])

BackgroundImage.propTypes = {
  /** background-image url */
  image: PropTypes.string,
  width: numberStringOrArray
}

BackgroundImage.defaultProps = {
  theme: theme
}

BackgroundImage.displayName = 'BackgroundImage'

export default BackgroundImage
