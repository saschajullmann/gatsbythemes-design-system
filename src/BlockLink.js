import styled from '@emotion/styled'
import Link from './Link'

const BlockLink = styled(Link)`
  display: block;
  color: inherit;
  text-decoration: none;
`

BlockLink.displayName = 'BlockLink'

export default BlockLink
