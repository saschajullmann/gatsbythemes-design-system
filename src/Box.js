import styled from '@emotion/styled'
import { space, width, color, textAlign } from 'styled-system'
import PropTypes from 'prop-types'
import theme from './theme'

const Box = styled.div`
  ${space} ${width} ${color} ${textAlign}
`

Box.displayName = 'Box'

Box.defaultProps = {
  theme: theme
}

Box.propTypes = {
  ...space.propTypes,
  ...width.propTypes,
  ...color.propTypes,
  ...textAlign.propTypes
}

export default Box
