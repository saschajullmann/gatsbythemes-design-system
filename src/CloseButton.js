import React from 'react'
import IconButton from './IconButton'
import PropTypes from 'prop-types'
import theme from './theme'

const CloseButton = props => <IconButton {...props} name="close" />

CloseButton.defaultProps = {
  size: 24,
  title: 'close'
}

CloseButton.propTypes = {
  onClick: PropTypes.func,
  size: PropTypes.number,
  title: PropTypes.string
}

CloseButton.displayName = 'CloseButton'

export default CloseButton
