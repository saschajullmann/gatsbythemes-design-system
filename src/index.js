export { default as Absolute } from './Absolute'
export { default as BackgroundImage } from './BackgroundImage'
export { default as Badge } from './Badge'
export { default as Banner } from './Banner'
export { default as BlockLink } from './BlockLink'
export { default as Box } from './Box'
export { default as Button } from './Button'
export { default as Card } from './Card'
export { default as CloseButton } from './CloseButton'
export { default as Container } from './Container'
export { default as Divider } from './Divider'
export { default as Flag } from './Flag'
export { default as Flex } from './Flex'
export { default as GreenButton } from './GreenButton'
export { default as Heading } from './Heading'
export { default as Hide } from './Hide'
export { default as Hug } from './Hug'
export { default as Icon } from './Icon'
export { default as IconButton } from './IconButton'
export { default as IconField } from './IconField'
export { default as Image } from './Image'
export { default as Input } from './Input'
export { default as InputField } from './FormField'
export { default as FormField } from './FormField'
export { default as InputGroup } from './InputGroup'
export { default as Label } from './Label'
export { default as Link } from './Link'
export { default as OutlineButton } from './OutlineButton'
export { default as Radio } from './Radio'
export { default as RatingBadge } from './RatingBadge'
export { default as RedButton } from './RedButton'
export { default as Relative } from './Relative'
export { default as Stamp } from './Stamp'
export { default as Select } from './Select'
export { default as Text } from './Text'
export { default as Tooltip } from './Tooltip'
export { default as ToggleBadge } from './ToggleBadge'
export { default as Truncate } from './Truncate'
export { default as Checkbox } from './Checkbox'

export { default as theme } from './theme'
export { default as ThemeProvider } from './ThemeProvider'

export * from './theme'