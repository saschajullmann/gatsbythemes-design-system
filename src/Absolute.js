import styled from '@emotion/styled'
import Box from './Box'
import { top, right, bottom, left, zIndex } from 'styled-system'

const Absolute = styled(Box)`
  position: absolute;
  ${top} ${bottom} ${left} ${right}
  ${zIndex}
`

Absolute.propTypes = {
  ...top.propTypes,
  ...right.propTypes,
  ...bottom.propTypes,
  ...left.propTypes,
  ...zIndex.propTypes
}

Absolute.displayName = 'Absolute'

export default Absolute
