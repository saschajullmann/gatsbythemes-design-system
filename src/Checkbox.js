import React from 'react'
import styled from '@emotion/styled'
import { css } from '@emotion/core'
import PropTypes from 'prop-types'
import theme from './theme'
import Icon from './Icon'
import Box from './Box'

const Checkbox = props => {
  const { disabled, size } = props
  return (
    <CheckBoxWrapper disabled={disabled}>
      <StyledInput type="checkbox" {...props} />
      <Icon name="boxChecked" size={size} data-name="checked" />
      <Icon name="boxEmpty" size={size} data-name="empty" />
    </CheckBoxWrapper>
  )
}

const CheckBoxWrapper = styled(Box)`
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: ${props =>
    props.disabled ? props.theme.colors.borderGray : props.theme.colors.gray};

  svg[data-name='checked'] {
    display: none;
  }

  > input:checked {
    & ~ svg[data-name='checked'] {
      display: inline-block;
      color: ${props =>
    props.disabled
      ? props.theme.colors.borderGray
      : props.theme.colors.blue};
    }

    & ~ svg[data-name='empty'] {
      display: none;
    }
  }
`

CheckBoxWrapper.defaultProps = {
  theme
}

const StyledInput = styled.input`
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;
`

Checkbox.displayName = 'Checkbox'

Checkbox.propTypes = {
  id: PropTypes.string.isRequired,
  size: PropTypes.number,
  onChange: PropTypes.func.isRequired
}

Checkbox.defaultProps = {
  size: 20
}

export default Checkbox
