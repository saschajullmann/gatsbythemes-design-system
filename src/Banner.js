import React from 'react'
import Box from './Box'
import Flex from './Flex'
import Text from './Text'
import Icon from './Icon'
import CloseButton from './CloseButton'
import Heading from './Heading'
import PropTypes from 'prop-types'

const bannerColors = {
  green: {
    backgroundColor: 'green',
    color: 'white',
    icon: 'success'
  },
  lightGreen: {
    backgroundColor: 'lightGreen',
    color: 'darkGreen',
    icon: 'success'
  },
  red: {
    backgroundColor: 'red',
    color: 'white',
    icon: 'warning'
  },
  lightRed: {
    backgroundColor: 'lightRed',
    color: 'darkRed',
    icon: 'warning'
  },
  orange: {
    backgroundColor: 'orange',
    color: 'white',
    icon: 'attention'
  },
  blue: {
    backgroundColor: 'blue',
    color: 'white',
    icon: 'information'
  },
  lightBlue: {
    backgroundColor: 'lightBlue',
    color: 'darkBlue',
    icon: 'information'
  }
}

const Banner = props => {
  const bannerColor = bannerColors[props.bg] || {}
  const icon = props.iconName || bannerColor.icon

  return (
    <Box
      {...props}
      bg={bannerColor.backgroundColor || props.bg}
      color={bannerColor.color || props.color}
    >
      <Flex justifyContent="space-between" alignItems="flex-start">
        {!!icon && !!props.showIcon && (
          <Icon name={icon} mr={2} size={24} mt="-2px" />
        )}
        <Box w={1}>
          <Text textAlign={props.textAlign}>
            <Heading.h5>{props.header}</Heading.h5>
            <Text.span fontSize={1}>{props.text}</Text.span>
            {props.children}
          </Text>
        </Box>
        {!!props.onClose && (
          <CloseButton
            onClick={props.onClose}
            ml={2}
            size={24}
            title="close"
            mt="-2px"
          />
        )}
      </Flex>
    </Box>
  )
}

Banner.displayName = 'Banner'

Banner.propTypes = {
  header: PropTypes.string,
  iconName: PropTypes.string,
  onClose: PropTypes.func,
  showIcon: PropTypes.bool,
  text: PropTypes.node,
  textAlign: PropTypes.string
}

Banner.defaultProps = {
  bg: 'green',
  textAlign: 'left',
  showIcon: true
}

export default Banner
