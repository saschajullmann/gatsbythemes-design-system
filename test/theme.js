import { theme, colors } from '../src'

const aliases = ['sm', 'md', 'lg', 'xl']

describe('theme', () => {
  console.warn = jest.fn()

  test('exports an object', () => {
    expect(typeof theme).toBe('object')
  })

  test('snapshot', () => {
    expect(theme).toMatchSnapshot()
  })

  test('exports colors', () => {
    expect(typeof colors).toBe('object')
  })

  test('breakpoints have aliases', () => {
    aliases.forEach((alias, i) =>
      expect(theme.breakpoints[alias]).toEqual(theme.breakpoints[i])
    )
  })

  test('media queries have aliases', () => {
    aliases.forEach((alias, i) =>
      expect(theme.mediaQueries[alias]).toEqual(theme.mediaQueries[i])
    )
  })
})
