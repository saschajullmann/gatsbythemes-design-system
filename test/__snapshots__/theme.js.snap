// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme snapshot 1`] = `
Object {
  "bold": 700,
  "boxShadows": Array [
    "0 0 2px 0 rgba(0,0,0,.08),0 1px 4px 0 rgba(0,0,0,.16)",
    "0 0 2px 0 rgba(0,0,0,.08),0 2px 8px 0 rgba(0,0,0,.16)",
    "0 0 2px 0 rgba(0,0,0,.08),0 4px 16px 0 rgba(0,0,0,.16)",
    "0 0 2px 0 rgba(0,0,0,.08),0 8px 32px 0 rgba(0,0,0,.16)",
  ],
  "breakpoints": Array [
    "32em",
    "40em",
    "48em",
    "64em",
  ],
  "colorStyles": Object {
    "blueOnLightBlue": Object {
      "backgroundColor": "#e8f2ff",
      "color": "#007aff",
    },
    "blueOnWhite": Object {
      "backgroundColor": "#fff",
      "color": "#007aff",
    },
    "danger": Object {
      "backgroundColor": "#c00",
      "color": "#fff",
    },
    "grayOnWhite": Object {
      "backgroundColor": "#fff",
      "color": "#4f6f8f",
    },
    "greenOnLightGreen": Object {
      "backgroundColor": "#ecf7ec",
      "color": "#0a0",
    },
    "greenOnWhite": Object {
      "backgroundColor": "#fff",
      "color": "#0a0",
    },
    "info": Object {
      "backgroundColor": "#f4f6f8",
      "color": "#001833",
    },
    "purpleOnLightPurple": Object {
      "backgroundColor": "#f5ebfa",
      "color": "#70b",
    },
    "purpleOnWhite": Object {
      "backgroundColor": "#fff",
      "color": "#70b",
    },
    "redOnLightRed": Object {
      "backgroundColor": "#fbebeb",
      "color": "#c00",
    },
    "redOnWhite": Object {
      "backgroundColor": "#fff",
      "color": "#c00",
    },
    "success": Object {
      "backgroundColor": "#0a0",
      "color": "#fff",
    },
    "textOnLightGray": Object {
      "backgroundColor": "#f4f6f8",
      "color": "#001833",
    },
    "textOnOrange": Object {
      "backgroundColor": "#f68013",
      "color": "#001833",
    },
    "textOnWhite": Object {
      "backgroundColor": "#fff",
      "color": "#001833",
    },
    "warning": Object {
      "backgroundColor": "#f68013",
      "color": "#001833",
    },
    "whiteOnBlue": Object {
      "backgroundColor": "#007aff",
      "color": "#fff",
    },
    "whiteOnDarkOrange": Object {
      "backgroundColor": "#f06f20",
      "color": "#fff",
    },
    "whiteOnGray": Object {
      "backgroundColor": "#4f6f8f",
      "color": "#fff",
    },
    "whiteOnGreen": Object {
      "backgroundColor": "#0a0",
      "color": "#fff",
    },
    "whiteOnPurple": Object {
      "backgroundColor": "#70b",
      "color": "#fff",
    },
    "whiteOnRed": Object {
      "backgroundColor": "#c00",
      "color": "#fff",
    },
    "whiteOnText": Object {
      "backgroundColor": "#001833",
      "color": "#fff",
    },
  },
  "colors": Object {
    "black": "#000",
    "blue": "#007aff",
    "borderGray": "#c0cad5",
    "darkBlue": "#049",
    "darkGray": "#364049",
    "darkGreen": "#060",
    "darkOrange": "#f06f20",
    "darkPurple": "#70b",
    "darkRed": "#800",
    "gray": "#4f6f8f",
    "green": "#0a0",
    "lightBlue": "#e8f2ff",
    "lightGray": "#f4f6f8",
    "lightGreen": "#ecf7ec",
    "lightOrange": "#f68013",
    "lightPurple": "#f5ebfa",
    "lightRed": "#fbebeb",
    "lightYellow": "#fff3c0",
    "orange": "#f68013",
    "purple": "#70b",
    "red": "#c00",
    "text": "#001833",
    "white": "#fff",
    "yellow": "#fedc2a",
  },
  "duration": Object {
    "fast": "150ms",
    "normal": "300ms",
    "slow": "450ms",
    "slowest": "600ms",
  },
  "font": "'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif",
  "fontSizes": Array [
    12,
    14,
    16,
    20,
    24,
    32,
    40,
    56,
    72,
  ],
  "fontWeights": Object {
    "bold": 700,
    "medium": 500,
    "regular": 500,
  },
  "letterSpacings": Object {
    "caps": "0.025em",
    "normal": "normal",
  },
  "lineHeights": Object {
    "display": 1.25,
    "standard": 1.5,
  },
  "maxContainerWidth": "1280px",
  "mediaQueries": Array [
    "@media screen and (min-width:32em)",
    "@media screen and (min-width:40em)",
    "@media screen and (min-width:48em)",
    "@media screen and (min-width:64em)",
  ],
  "radii": Array [
    0,
    2,
    6,
  ],
  "radius": "2px",
  "regular": 500,
  "space": Array [
    0,
    4,
    8,
    16,
    32,
    64,
    128,
  ],
  "textStyles": Object {
    "body0": Object {
      "fontSize": "12px",
      "fontWeight": 500,
      "lineHeight": 1.5,
    },
    "body1": Object {
      "fontSize": "14px",
      "fontWeight": 500,
      "lineHeight": 1.5,
    },
    "body2": Object {
      "fontSize": "16px",
      "fontWeight": 500,
      "lineHeight": 1.5,
    },
    "display0": Object {
      "fontSize": "12px",
      "fontWeight": 700,
      "letterSpacing": "0.025em",
      "lineHeight": 1.25,
      "textTransform": "uppercase",
    },
    "display1": Object {
      "fontSize": "14px",
      "fontWeight": 700,
      "lineHeight": 1.25,
    },
    "display2": Object {
      "fontSize": "16px",
      "fontWeight": 700,
      "lineHeight": 1.25,
    },
    "display3": Object {
      "fontSize": "20px",
      "fontWeight": 700,
      "lineHeight": 1.25,
    },
    "display4": Object {
      "fontSize": "24px",
      "fontWeight": 700,
      "lineHeight": 1.25,
    },
    "display5": Object {
      "fontSize": "32px",
      "fontWeight": 700,
      "lineHeight": 1.25,
    },
    "display6": Object {
      "fontSize": "40px",
      "fontWeight": 700,
      "lineHeight": 1.25,
    },
    "display7": Object {
      "fontSize": "56px",
      "fontWeight": 700,
      "lineHeight": 1.25,
    },
    "display8": Object {
      "fontSize": "72px",
      "fontWeight": 700,
      "lineHeight": 1.25,
    },
  },
  "timingFunctions": Object {
    "easeIn": "cubic-bezier(0.5, 0, 1, 1)",
    "easeInOut": "cubic-bezier(0.5, 0, 0.25, 1)",
    "easeOut": "cubic-bezier(0, 0, 0.25, 1)",
  },
  "transitionDelays": Object {
    "large": "260ms",
    "medium": "160ms",
    "small": "60ms",
    "xLarge": "360ms",
  },
}
`;
